packages/@webex/plugin-cc/src/services/task/index.ts

```diff
@@ -0,0 +1,83 @@
+import {LocalMicrophoneStream} from '@webex/calling';
+import {getErrorDetails} from '../core/Utils';
+import {LoginOption} from '../../types';
+import {CC_FILE} from '../../constants';
+import routingContact from './contact';
+import {ITask, TaskId, TaskResponse, TaskData} from './types';
+import WebCallingService from '../WebCallingService';
+
+export default class Task implements ITask {
+  private contact: ReturnType<typeof routingContact>;
+  private localAudioStream: LocalMicrophoneStream;
+  private webCallingService: WebCallingService;
+  public data: TaskData;
+
+  constructor(
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    data: TaskData
+  ) {
+    this.contact = contact;
+    this.data = data;
+    this.webCallingService = webCallingService;
+  }
+
+  /**
+   * This is used for incoming task accept by agent.
+   *
+   * @param taskId - Unique Id to identify each task
+   *
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.accept(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async accept(taskId: TaskId): Promise<TaskResponse> {
+    try {
+      if (this.webCallingService.loginOption === LoginOption.BROWSER) {
+        const constraints = {
+          audio: true,
+        };
+
+        const localStream = await navigator.mediaDevices.getUserMedia(constraints);
+        const audioTrack = localStream.getAudioTracks()[0];
+        this.localAudioStream = new LocalMicrophoneStream(new MediaStream([audioTrack]));
+        this.webCallingService.answerCall(this.localAudioStream, taskId);
```

Comment:
The answer and decline calls seem to suggest that there will only be one call always.. are we sure this is always the case?

Reply:
Actually, IIRC once an agent answers a call they go into a different state so no other call should arrive.. so maybe my scenario won't happen.. but we should try to link the call and the task somehow.

Reply:
There will be one call at a time